src/main/kotlin/se/uulm/snowballr/backend/db/Database.kt

```diff
@@ -125,29 +125,40 @@ class Database(
     }
 
     /**
-     * Initializes a dummy user in the database if the environment variable is set to use a dummy user
-     * and the dummy user does not already exist.
+     * Sets up a dummy user in the database if the environment configuration requires it.
      *
-     * The dummy user is created with predefined credentials and role, which can be used for testing purposes.
+     * This method checks if a dummy user is needed based on the environment settings. If so, it
+     * creates the dummy user if it does not already exist. If the dummy user is not needed, it
+     * deletes the existing dummy user from the database.
      */
-    fun useDummyUser() {
-        if (!envReader.env.miscellaneous.useDummyUser) {
-            return
-        }
+    fun setupDummyUser() {
+        val useDummyUser = envReader.env.miscellaneous.useDummyUser
 
         val existingId = UserTable
-            .selectAll()
+            .select(UserTable.id)
             .where { UserTable.email eq DummyUser.email }
-            .map { it.toUser().id }
+            .map { it[UserTable.id].value }
             .singleOrNull()
 
-        DummyUser.id = existingId ?: UserTable.insertAndGetId {
-            it[email] = DummyUser.email
-            it[firstName] = DummyUser.firstName
-            it[lastName] = DummyUser.lastName
-            it[passwordHash] = DummyUser.passwordHash
-            it[role] = DummyUser.role
-            it[status] = DummyUser.status
-        }.value
+        if (useDummyUser) {
+            // If the dummy user is needed, create it if it does not exist
+            DummyUser.id = existingId ?: UserTable.insertAndGetId {
+                it[email] = DummyUser.email
+                it[firstName] = DummyUser.firstName
+                it[lastName] = DummyUser.lastName
+                it[passwordHash] = DummyUser.passwordHash
+                it[role] = DummyUser.role
+                it[status] = DummyUser.status
+            }.value
+
+            logger.info { "Dummy User ID: ${DummyUser.id}, Password: ${DummyUser.password}" }
+        } else {
+            // If the dummy user is not needed, delete it if it exists
+            if (existingId != null) {
+                UserTable.deleteWhere { UserTable.id eq existingId }.also {
```

Comment:
By using `else if` you can safe on stage of nesting.
```suggestion
} else if (existingId != null) {
    // If the dummy user is not needed, delete it if it exists
    UserTable.deleteWhere { UserTable.id eq existingId }.also {
```